tests/storage/cdi_upload/test_upload_virtctl.py

```
@@ -53,6 +55,30 @@ def skip_no_reencrypt_route(upload_proxy_route):
+    and return localhost URL instead of the route (which resolves to IPv4).
+    """
+    if is_ipv6_single_stack_cluster():
+        port = 8443
+        port_forward_process = subprocess.Popen(
+            ["oc", "port-forward", f"svc/{CDI_UPLOADPROXY}", f"{port}:443", "-n", hco_namespace.name],
+        )
+        LOGGER.info(f"Started port-forward (PID={port_forward_process.pid}) to {CDI_UPLOADPROXY}")
+
+        yield f"https://localhost:{port}"
+
+        LOGGER.info(f"Terminating port-forward process (PID={port_forward_process.pid})")
```

Ruff (0.15.2)

[warning] Logging statement uses f-string

(G004)